### PR TITLE
feat: Entropyfi integration

### PR DESCRIPTION
**Pull request by billxljin**
Entropyfi Mask plugin integration
-----

### What Is Entropyfi?

Entropyfi is a crowd wisdom machine built on the lossless prediction market. Prediction market has long been theorized as a way to extract the wisdom of the crowd, yet none has succeed at scaling the process. Entropyfi differentiates itself with its lossless prediction market, which removes the financial disincentive that penalizes users who make wrong predictions. Instead, Entropyfi utilizes the yield on deposit pool to reward winners. 

With a better aligned financial incentive design, Entropyfi is able to scale the prediction market and provide more reliable and scalable crowd wisdom sourcing machine. Entropyfi’s goal is to make the crowd wisdom accessible to the everyone, such that novel financial derivatives could be built on top of Entropyfi.

Website: https://entropyfi.com/
[Entropyfi Mask integration function check list 2021-09-23.pdf](https://github.com/DimensionDev/Maskbook/files/7253064/Entropyfi.Mask.integration.function.check.list.2021-09-23.pdf)

### CODE REVIEW

File: packages/maskbook/src/plugins/Entropyfi/hooks/useToggle.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { useCallback, useState } from 'react'
+
+export default function useToggle(initialState = false): [boolean, () => void] {
```
Comment (Jack-Works) [inline packages/maskbook/src/plugins/Entropyfi/hooks/useToggle.ts:3]: Use [useToggle](https://github.com/streamich/react-use/blob/master/docs/useToggle.md) from `react-use`.

File: packages/maskbook/src/plugins/Entropyfi/base.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    ENTROPYFI_PLUGIN_NAME,
+    ENTROPYFI_PLUGIN_DESCRIPTION,
+} from './constants'
+
+export const base: Plugin.Shared.Definition = {
+    ID: ENTROPYFI_PLUGIN_ID,
+    icon: ENTROPYFI_PLUGIN_ICON,
+    name: { fallback: ENTROPYFI_PLUGIN_NAME },
+    description: { fallback: ENTROPYFI_PLUGIN_DESCRIPTION },
+    publisher: { name: { fallback: 'Mask Network' }, link: 'https://mask.io/' },
```
Comment (Jack-Works) [inline packages/maskbook/src/plugins/Entropyfi/base.ts:15]: Please change it to your team.

File: packages/maskbook/src/plugins/Entropyfi/UI/NetworkView.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        alignItems: 'center',
+    },
+    icon: {
+        width: '1em',
+        height: '1em',
+        marginRight: theme.spacing(0.5),
+    },
+}))
+
+interface NetworkViewProps extends withClasses<never> {
```
Comment (Jack-Works) [inline packages/maskbook/src/plugins/Entropyfi/UI/NetworkView.tsx:20]:
```suggestion
interface NetworkViewProps {
```

File: .eslintrc.json [hunk 4]
```diff
@@ -31,6 +31,12 @@
         "prefer-const": "warn",
         "use-isnan": "error",
         "unused-imports/no-unused-imports-ts": "error",
-        "lodash/import-scope": ["error", "member"]
+        "lodash/import-scope": ["error", "member"],
+        "prettier/prettier": [
+            "error",
+            {
+                "endOfLine": "auto"
```
Comment (Jack-Works) [inline .eslintrc.json:38]: Hi, please don't adjust this. Did you encounter problems with CR/LF?

**Review by septs — CHANGES_REQUESTED** [review]
plz use https://aka.pw/svgomg reduce all svg image resources.

File: .eslintrc.json (hunk 4 above)
Comment (septs) [reply .eslintrc.json:38]: plz revert the code style changes, thanks

**Comment by AinuX** [comment]
Once it's available for testing, pleaes notify me.

**Review by septs — CHANGES_REQUESTED** [review]
plz do not use `any` type

File: packages/maskbook/src/plugins/Entropyfi/UI/Component/Card_Left.tsx [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+            setter(num)
+            if (num >= value) {
+                clearInterval(timer)
+                setter(value)
+            }
+        }, speed)
+    } else setter(value)
+}
+
+const colorChange = (setter: any, speed: number, cvalue: string) => {
```
Comment (septs) [inline packages/maskbook/src/plugins/Entropyfi/UI/Component/Card_Left.tsx:170]:
```suggestion
const colorChange = (setter: (r: number, g: number, b: number) => void, speed: number, cvalue: string) => {
```

**Comment by billxljin** [comment]
Thank you for your kind suggestions! We have modified the code, please review them again.

File: packages/maskbook/src/plugins/Entropyfi/UI/Component/Card_Right.tsx [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+                </Grid>
+            )}
+        </Grid>
+    )
+}
+
+const TIPS = (props: any) => {
+    return (
+        <div style={{ opacity: props.show ? 1 : 0 }}>
+            <p>
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Entropyfi/UI/Component/Card_Right.tsx:268]: Texts should be wrapped by `Typography`.

File: packages/maskbook/src/plugins/Entropyfi/constants/assets/global_usdt.tsx [hunk 7]
```diff
@@ -0,0 +1,18 @@
+import { SvgIcon, SvgIconProps } from '@material-ui/core'
+import type { FC } from 'react'
+
+export const UsdtIcon: FC<SvgIconProps> = (props) => (
+    <SvgIcon {...props}>
+        <svg viewBox="0 0 1024 1024">
+            <path
+                fill="#26A17B"
+                d="M0 512a512 512 0 1 0 1024 0A512 512 0 1 0 0 512z"
+                data-spm-anchor-id="a313x.7781069.0.i12"
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Entropyfi/constants/assets/global_usdt.tsx:10]:
```suggestion
```

File: packages/maskbook/src/plugins/Entropyfi/constants/assets/card_wait.tsx [hunk 8]
```diff
@@ -0,0 +1,19 @@
+import { SvgIcon, SvgIconProps } from '@material-ui/core'
+import type { FC } from 'react'
+
+export const WaitIcon: FC<SvgIconProps> = (props) => (
+    <SvgIcon {...props}>
+        <svg viewBox="0 0 1024 1024">
+            <path
+                fill="#878fa5"
+                d="M262 138h474L597 354l88 51 187-291-44-75H161l-43 78 321 422 286 359H252l139-215-88-52-186 291 44 75h667l42-79-347-437z"
+                data-spm-anchor-id="a313x.7781069.0.i0"
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Entropyfi/constants/assets/card_wait.tsx:10]:
```suggestion
```

File: packages/maskbook/src/plugins/Entropyfi/constants.ts [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+        [poolId: string]: {
+            [tokenType: string]: Token
+        }
+    }
+}
+
+export const tokenMap: TokenMap = {
+    42: {
+        'BTC-USDT': {
+            principalToken: new Token(42, '0x13512979ADE267AB5100878E2e0f485B568328a4', 6, 'USDT', 'USDT Coin'),
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Entropyfi/constants.ts:103]: Move these addresses to `web3-constants`?

File: packages/maskbook/src/plugins/Entropyfi/UI/Component/Card_Left.tsx [hunk 10]
```diff
@@ -0,0 +1,193 @@
...
+
+    // console.log('props.poolId:', props.poolId)
+    // console.log('PoolView.chainId:', chainId)
+    // console.log('shortValue:', shortValue)
+    // console.log('longValue:', longValue)
+
+    const poolValue = new BigNumber(shortValue)
+        .plus(new BigNumber(longValue))
+        .plus(new BigNumber(sponsorValue ? sponsorValue.toExact() : '0'))
+        .toFixed(0)
```
Comment (septs) [inline packages/maskbook/src/plugins/Entropyfi/UI/Component/Card_Left.tsx:95]:
```suggestion
        .plus(longValue)
        .plus(sponsorValue ? sponsorValue.toExact() : '0')
        .toFixed(0)
```

**Pull request closed without merging**